extensions/oidc/runtime/src/main/java/io/quarkus/oidc/Tenant.java

```diff
@@ -9,7 +10,7 @@
 /**
  * Qualifier which can be used to associate one or more OIDC features with a named tenant.
  */
-@Target({ TYPE })
+@Target({ TYPE, METHOD })
```

Comment:
It already introduces a problem if someone adds it to `TokenHeaderCustomizer` implementation method instead of class.
I should've named this annotation `TenantFeature` which aligns better with the current docs: `... associate one or more OIDC features with a named tenant.`.

Hmm... `TokenHeaderCustomizer` is a new feature in 3.2 but for the only case where it was required so far, Azure token customizations, we ship `AzureTokenCustomizer` which can be referred to from tenant configurations with the `tenant-customizer-name` property and it is very unlikely that users have used `@Tenant` to apply some other custom token customizers.

My proposal would be to have `@TenantFeature` annotation used for what current `@Tenant` (and update token customizer related JavaDocs and tests to refer to `@TenantFeature`) and mark it as a breaking change, though I'd be nearly certain it won't break any user's code. I'll add a migration note. 

And your PR then keeps working with `@Tenant` but updates its docs to say it can be used to `associate tenant configurations with JAX-RS classes and methods`, and we will evolve both annotations as required.

Or we can add a new annotation like `@TenantId` right now to support this PR, but I feel `@TenantFeature` would be a better name for what `@Tenant` was originally meant to - associate whatever feature it marks with some tenant.

Does it sound reasonable ? What do you think ?

Reply:
> It already introduces a problem if someone adds it to TokenHeaderCustomizer implementation method instead of class.

We disagree on that point, but IMHO it is important that you must be fine with all changes here, so let's avoid using one annotation for token customizer and Jakarta REST resources :-)

It seems to have both `@Tenant` and `@TenantId` would be confusing (though it can be documented), `@TenantFeature` seems much better to me. I can wait with this PR till you do changes (I take it from context you want to do it, right? I can do it if you are busy).

Reply:
@michalvavrik Hey, let me keep this one open for Georgios and others to have a look at what we've discussed, when reviewing #34862